packages/react/src/hooks/web3inboxClient.ts

```diff
@@ -63,27 +63,23 @@ export const useWeb3InboxClient = () => {
 
 export const useW3iAccount = () => {
   const client = useWeb3InboxClient();
-  const [identityKey, setIdentityKey] = useState<string | null>(null);
+  const [isRegistered, setIsRegistered] = useState<boolean>(false);
 
-  const { account } = useClientState();
+  const { account, registration } = useClientState();
 
   const setAccount = useCallback(
     (account: string) => {
```

Comment:
Can we use `async/await` and add another state variable for `isRegistering` ?
```suggestion
    async (account: string) => {
```

Reply:
made the function async and return the promise, and added `isRegistering` to the register method.